### PR TITLE
Add tests and cast embeds_one and embeds_many

### LINKED ISSUE #195: Doesn't work with embeds
Embeds are broken, seems like #73 was overwritten, is this something that could be added back in?

```Factory.insert(:user)
** (RuntimeError) changing embeds with change/2 or put_change/3 is not supported, please use put_embed/4 instead
      (ecto) lib/ecto/changeset.ex:1011: Ecto.Changeset.put_change/7
    (stdlib) lists.erl:1263: :lists.foldl/3
      (ecto) lib/ecto/changeset.ex:325: Ecto.Changeset.change/2
      (ecto) lib/ecto/changeset/relation.ex:152: Ecto.Changeset.Relation.do_change/4
      (ecto) lib/ecto/changeset/relation.ex:237: Ecto.Changeset.Relation.single_change/5
      (ecto) lib/ecto/repo/schema.ex:516: anonymous fn/4 in Ecto.Repo.Schema.surface_changes/4
    (elixir) lib/enum.ex:1755: Enum."-reduce/3-lists^foldl/2-0-"/3
      (ecto) lib/ecto/repo/schema.ex:503: Ecto.Repo.Schema.surface_changes/4
```

### PR DESCRIPTION
**Pull request by paulcsmith**
This closes #195

### CODE REVIEW

File: test/ex_machina/ecto_strategy_test.exs [hunk 1]
```diff
@@ -59,6 +59,19 @@ defmodule ExMachina.EctoStrategyTest do
     assert List.first(model.articles).visits == Decimal.new(10)
   end
 
+  test "insert/1 casts embedded associations" do
+    author = %ExMachina.Author{name: "Paul", salary: Decimal.new(10.3)}
```
Comment (jsteiner) [inline test/ex_machina/ecto_strategy_test.exs]: Should we test `salary: 10.3` to ensure it casts it to a decimal?

Comment (paulcsmith) [inline test/ex_machina/ecto_strategy_test.exs]: Good idea!

**Comment by paulcsmith** [comment]
Anything else or does this look good @jsteiner?

File: test/ex_machina/ecto_strategy_test.exs [hunk 2]
```diff
@@ -59,6 +59,19 @@ defmodule ExMachina.EctoStrategyTest do
...
+  test "insert/1 casts embedded associations" do
+    author = %ExMachina.Author{name: "Paul", salary: 10.3}
+    link = %ExMachina.Link{url: "wow"}
+
+    comment = TestFactory.insert :comment_with_embedded_assocs,
+      author: author,
+      links: [link]
+
+    assert comment.author.name == author.name
+    assert comment.author.salary == author.salary
```
Comment (jsteiner) [inline test/ex_machina/ecto_strategy_test.exs]: I think we want to test that this becomes `Decimal.new(salary)`

Comment (paulcsmith) [inline test/ex_machina/ecto_strategy_test.exs]: Derp. Thanks. That means it probably isn't casting. I'll fix the test and add the corresponding code

**Pull request merged**